### ISSUE TITLE
worker.mixin.RemoteMethodAccess: accessing new main threads too early

### ISSUE THREAD

**Issue by tobiu** [neomjs/neo#5577]
> **Current state, 2026-09-15.** Read at `dev@624ae90841`. This body is the record; the thread below it is history. The 2024 report and its two options are preserved under *Origin*.

## The defect

The app worker calls a main-thread remote before that thread has registered it. The 2024 sighting was theme-file loading against a newly opened window in the colors websocket demo: `Neo.main.addon.Stylesheet.addThemeFiles(…)` reaches a main thread whose `Stylesheet` addon has not registered, and the call rejects with an invalid-namespace error.

## What already shipped

`Main#onDomContentLoaded` is deterministic today (#7815 lineage):

```js
const instances = modules.map(module => me.registerAddon(module.default));

await Promise.all(instances.map(instance => instance.remotesReady()));
await me.remotesReady();

WorkerManager.onWorkerConstructed({origin: 'main'})
```

`core.Base#remotesReady()` resolves inside `initAsync`, after `initRemote()` has sent the registrations. So a window's boot addons are registered before that window reports itself constructed — the 2024 option 1, for the boot path.

## What this ticket fixes

**The runtime import path had no such gate.** `Main#importAddon` — called by `worker.App#importAddon` and by `ai.client.InteractionService` — created the addon and slept:

```js
this.registerAddon(module.default);
await this.timeout(20); // Wait until remotes are registered
```

The caller uses the addon's proxy the moment that promise settles, and registration is a round trip through `core.Base#initRemote` — no fixed delay bounds a round trip. That is the same defect as the 2024 sighting, on the path that stayed unguarded. It now awaits `registerAddon(...).remotesReady()`, the promise the boot path waits on.

**Registration is not readiness.** `main.addon.Base#initAsync` awaits `super.initAsync()`, which resolves `remotesReady`, and only afterwards its `#loadFilesPromise`. So Monaco, Mermaid, AmCharts and OpenStreetMaps register early and merely stay `isReady: false` while their libraries load — a lazily loading addon may never reach readiness at all. The caller wants the proxy, not the library, so `remotesReady()` is the gate and `ready()` would hang. (First recorded here with the two confused; corrected after @neo-gpt-emmy's R1 on the PR.)

## The fixed delays that remain, and why

Named here so the next reader does not re-derive them:

| site | what it guards | can it become deterministic today |
|---|---|---|
| `worker.Manager#onWorkerConstructed`, `loadApplicationDelay` (20 ms), commented *"better safe than sorry => all remotes need to be registered"* | the window's workers having registered **every** remote, not just their own worker singleton's | **No.** A worker announces `workerConstructed` from `onConstructed`, synchronously, while each of its singletons schedules `initRemote` in its own microtask. 50 source files declare a `remote` config; there is no worker-wide registry of pending `initAsync` work, so nothing can be awaited. A gate needs that registry first — a design change, not a substitution. |
| `worker.Base#onConnect` and `worker.App#onConnect`, `await this.timeout(10)` | the app's view controllers being in place before `connect` fires | Out of scope: a different concern from remote registration, and its own readiness signal. |

## Acceptance Criteria

- [ ] AC-1 — An addon imported at runtime has its remotes registered before `Main#importAddon` resolves, and the arm separates that from readiness: a real `main.addon.Base` whose files never load and whose `initRemote` is delayed must resolve the import with `registered: true` and `isReady: false`, with an instantly registering addon as the control that the arm reads state rather than elapsed time.
- [ ] AC-2 — The arm fails against both neighbouring gates. *(Mutations: the restored 20 ms sleep resolves before registration, `registered: false`; `ready()` in place of `remotesReady()` never resolves at all.)*
- [ ] AC-3 — The remaining fixed delays are recorded above with what each guards and what a deterministic replacement would require, so this ticket does not close as if the whole class were gone.

## Origin (2024-07-15, @tobiu)

> It is an edge-case bug which does happen inside our colors app websocket demo: the app worker tries to send messages to the new main thread => loading theme files, before these remotes have been registered.
>
> 2 options:
> 1. debug the "app is ready" => connect event to ensure it fires once all main thread addons are ready.
> 2. if a namespace does not exist yet, try again 100(?)ms later

Option 1 shipped for the boot path and is completed here for the runtime path. Option 2 is not taken: retrying a call whose namespace is absent turns a race into a slower race, and the readiness promise it would poll for already exists.

## Related

- #7815 — the boot-path gate that replaced the first sleep
- PR for this ticket: the runtime path

**Comment by tobiu** [comment]
this one must not auto close.
